packages/forms/resources/views/components/text-input.blade.php

```diff
@@ -60,6 +63,7 @@ class="fi-fo-text-input"
                         $applyStateBindingModifiers('wire:model') => $statePath,
                         'x-data' => (count($extraAlpineAttributes) || filled($mask)) ? '{}' : null,
```

Comment:
Sorry for missing this initially, but the `x-data` in here should be used instead of the one added in this PR?

Reply:
In that case the icon button won't be able to change the state of `inputType` variable as it is outside the text input scope (it's inside the wrapper, not a child of input component)